B/bitshuffle/build_tarballs.jl

```diff
@@ -0,0 +1,39 @@
+# Note that this script can accept some limited command-line arguments, run
+# `julia build_tarballs.jl --help` to see a usage message.
+using BinaryBuilder, Pkg
+
+name = "bitshuffle"
+version = v"0.4.2"
+
+# Collection of sources required to complete build
+sources = [
+    GitSource("https://github.com/kiyo-masui/bitshuffle.git", "a60471d37a8cbbd8265dc8cfa83a9320abdcb590"),
+    GitSource("https://github.com/facebook/zstd.git", "e47e674cd09583ff0503f0f6defd6d23d8b718d3")
+]
+
+# Bash recipe for building across all platforms
+script = raw"""
+cd $WORKSPACE/srcdir/bitshuffle
+mkdir -p "${libdir}"
+cc -O3 -std=c99 -DZSTD_SUPPORT -Ilz4 -Isrc -I../zstd/lib -lzstd -llz4 -fPIC --shared -o "${libdir}/libbitshuffle.${dlext}" src/bitshuffle.c src/iochain.c src/bitshuffle_core.c
+"""
+
+# These are the platforms we will build for by default, unless further
+# platforms are passed in on the command line
+platforms = supported_platforms(exclude=[Platform("i686","windows"),
+					  Platform("x86_64","windows")
+					  ])
```

Comment:
What's the problem with Windows?  Also, if anything `exclude=Sys.iswindows` is much simpler.

Reply:
The problem is that the mingw32 compiler tries to resolve lz4 library symbols in the shared library being built and fails. The same command-line options are fine on linux and mac. I assume there are some windows-specific command line options for building shared libraries that depend on other shared libraries but I haven't found them yet.

Reply:
I can't tell more without seeing the precise error messages.

Reply:
Here is the full error message for target `x86_64-w64-mingw32`:

```[12:18:34]  ---> cc -O3 -std=c99 -DZSTD_SUPPORT -I${includedir} -Isrc -lzstd -llz4 -fPIC --shared -o "${libdir}/libbitshuffle.${dlext}" src/bitshuffle.c src/iochain.c src/bitshuffle_core.c
[12:18:34] src/bitshuffle.c:1:0: warning: -fPIC ignored for target (all code is position independent) [enabled by default]
[12:18:34]  /*
[12:18:34]  ^
[12:18:34] src/iochain.c:1:0: warning: -fPIC ignored for target (all code is position independent) [enabled by default]
[12:18:34]  /*
[12:18:34]  ^
[12:18:34] src/bitshuffle_core.c:1:0: warning: -fPIC ignored for target (all code is position independent) [enabled by default]
[12:18:34]  /*
[12:18:34]  ^
[12:18:34] /tmp/ccNHMMif.o:bitshuffle.c:(.text+0x42): undefined reference to `LZ4_compressBound'
[12:18:34] /tmp/ccNHMMif.o:bitshuffle.c:(.text+0xaf): undefined reference to `LZ4_compress_default'
[12:18:34] /tmp/ccNHMMif.o:bitshuffle.c:(.text+0x20a): undefined reference to `LZ4_decompress_safe'
[12:18:34] /tmp/ccNHMMif.o:bitshuffle.c:(.text+0x2dd): undefined reference to `ZSTD_compressBound'
[12:18:34] /tmp/ccNHMMif.o:bitshuffle.c:(.text+0x350): undefined reference to `ZSTD_compress'
[12:18:34] /tmp/ccNHMMif.o:bitshuffle.c:(.text+0x4a5): undefined reference to `ZSTD_decompress'
[12:18:34] /tmp/ccNHMMif.o:bitshuffle.c:(.text+0x594): undefined reference to `LZ4_compressBound'
[12:18:34] /tmp/ccNHMMif.o:bitshuffle.c:(.text+0x5d6): undefined reference to `LZ4_compressBound'
[12:18:34] /tmp/ccNHMMif.o:bitshuffle.c:(.text+0x6c5): undefined reference to `ZSTD_compressBound'
[12:18:34] /tmp/ccNHMMif.o:bitshuffle.c:(.text+0x708): undefined reference to `ZSTD_compressBound'
[12:18:34] /opt/x86_64-w64-mingw32/bin/../lib/gcc/x86_64-w64-mingw32/4.8.5/../../../../x86_64-w64-mingw32/bin/ld: /tmp/ccNHMMif.o: bad reloc address 0x0 in section `.pdata'
[12:18:34] collect2: error: ld returned 1 exit status
[12:18:34]  ---> cc -O3 -std=c99 -DZSTD_SUPPORT -I${includedir} -Isrc -lzstd -llz4 -fPIC --shared -o "${libdir}/libbitshuffle.${dlext}" src/bitshuffle.c src/iochain.c src/bitshuffle_core.c
[12:18:34] Previous command exited with 1
[12:18:34] Child Process exited, exit code 1
```

Reply:
The problem seems to be order of the arguments. The source files should be to the left of the link parameter.

Reply:
Thanks for figuring that out!